widget/button.go

```diff
@@ -78,12 +78,8 @@ func (b *buttonRenderer) Refresh() {
 	b.label.Text = b.button.Text
 
 	if b.button.Icon != nil {
-		if b.icon == nil {
-			b.icon = canvas.NewImageFromResource(b.button.Icon)
-			b.objects = append(b.objects, b.icon)
-		} else {
-			b.icon.Resource = b.button.Icon
-		}
+		b.icon = canvas.NewImageFromResource(b.button.Icon)
+		b.objects = append(b.objects, b.icon)
```

Comment:
I think that every time we refresh the button a new icon will be added to the object list